### PR TITLE
fix shellcheck failures in hack/test-integration.sh and hack/verify-test-featuregates.sh

### PR DESCRIPTION
**Pull request by xichengliudui**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:

Ref #72956
/kind cleanup
/sig testing
/priority important-soon

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: hack/verify-test-featuregates.sh [hunk 1]
```diff
@@ -26,7 +26,7 @@ cd "${KUBE_ROOT}"
 rc=0
 
 # find test files accessing the mutable global feature gate or interface
-direct_sets=$(grep -n --include '*_test.go' -R 'MutableFeatureGate' . 2>/dev/null) || true
+direct_sets=$(grep -n --include '.*_test.go' -R 'MutableFeatureGate' . 2>/dev/null) || true
```
Comment (xichengliudui) [inline hack/verify-test-featuregates.sh]: ref: https://github.com/koalaman/shellcheck/wiki/SC2063

File: hack/test-integration.sh [hunk 2]
```diff
@@ -20,8 +20,6 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
```
Comment (xichengliudui) [inline hack/test-integration.sh]: I found that KUBE_ROOT was not used in the script, so I removed it.
ref: https://github.com/koalaman/shellcheck/wiki/SC2034

File: hack/verify-test-featuregates.sh (hunk 1 above)
Comment (k-toyoda-pi) [reply hack/verify-test-featuregates.sh]: I think that it cannot use regex with `--include`. If `.*_test.go`, `grep` does not include any files. See #76855. I proposed another way to fix this line in #80505.

Comment (xichengliudui) [reply hack/verify-test-featuregates.sh]: thinks.

**Comment by BenTheElder** [comment]
/retitle fix shellcheck failures in hack/test-integration.sh and hack/verify-test-featuregates.sh

**Review by BenTheElder — APPROVED** [review]
/lgtm
/approve
thanks! needs a rebase

**Pull request closed without merging**